src/Element/ExecutablePluginSelect.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\commerce\Element;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\Core\Render\Element\FormElement;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @FormElement("commerce_executable_plugin_select")
+ */
+class ExecutablePluginSelect extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    return [
+      '#input' => TRUE,
+      '#plugin_type_id' => NULL,
+      '#process' => [
+        [$class, 'processExecutablePluginSelect'],
+        [$class, 'processAjaxForm'],
+      ],
+    ];
+  }
+
+  /**
+   * Process callback.
+   */
+  public static function processExecutablePluginSelect(&$element, FormStateInterface $form_state, &$complete_form) {
+    if (!$element['#plugin_type_id']) {
+      throw new \InvalidArgumentException('You must specify the plugin type ID.');
+    }
+
+    $element['#tree'] = TRUE;
+
+    /** @var \Drupal\Core\Executable\ExecutableManagerInterface $executable_manager */
+    $executable_manager = \Drupal::service('plugin.manager.' . $element['#plugin_type_id']);
+
+    $values = $element['#value'];
+
+    $executable_id = !empty($values['executable_id']) ? $values['executable_id'] : '_none';
+
+    $ajax_wrapper_id = Html::getUniqueId('ajax-wrapper');
+    $ajax_settings = [
+      'callback' => [get_called_class(), 'executableIdAjax'],
+      'wrapper' => $ajax_wrapper_id,
+      'options' => [
+        'query' => [
+          'element_parents' => implode('/', $element['#array_parents']),
+        ],
+      ],
+    ];
+
+    // Prefix and suffix used for Ajax replacement.
+    $element['#prefix'] = '<div id="' . $ajax_wrapper_id . '">';
+    $element['#suffix'] = '</div>';
+
+    $element['executable_id'] = [
+      '#type' => 'select',
+      '#title' => t('Select plugin'),
```

Comment:
We probablay want to be able to override this #title? Perhaps by just mirroring $element['#title']?